server/src-lib/Hasura/RQL/DDL/RemoteSchema.hs

```diff
@@ -52,6 +56,55 @@ runAddRemoteSchema env q = do
   where
     name = _arsqName q
 
+runAddRemoteSchemaPermissions
+  :: ( QErrM m
+     , CacheRWM m
+     , MonadTx m
+     , HasEnableRemoteSchemaPermsCtx m
+     )
+  => AddRemoteSchemaPermissions
+  -> m EncJSON
+runAddRemoteSchemaPermissions q = do
+  isRSPermsEnabled <- enableRemoteSchemaPerms <$> askEnableRemoteSchemaPermsCtx
+  unless isRSPermsEnabled $ do
+    throw400 ConstraintViolation
+      $ "remote schema permissions can only be added when "
+      <> "remote schema permissions are enabled in the graphql-engine"
+  remoteSchemaMap <- scRemoteSchemas <$> askSchemaCache
+  (RemoteSchemaCtx _ ctx perms) <-
+    onNothing (Map.lookup name remoteSchemaMap) $
+      throw400 NotExists $ "remote schema " <> name <<> " doesn't exist"
+  onJust (Map.lookup role perms) $ \_ ->
+    throw400 AlreadyExists $ "permissions for role: " <> role <<> " for remote schema:"
+      <> name <<> " already exists"
+  resolveRoleBasedRemoteSchema providedSchemaDoc ctx
+  liftTx $ addRemoteSchemaPermissionsToCatalog q
+  buildSchemaCacheFor $ MORemoteSchemaPermissions name role
+  pure successMsg
+  where
+    AddRemoteSchemaPermissions name role defn _ = q
+
+    providedSchemaDoc = _rspdSchema defn
+
+runDropRemoteSchemaPermissions
+  :: ( QErrM m
+     , CacheRWM m
+     , MonadTx m
+     )
+  => DropRemoteSchemaPermissions
+  -> m EncJSON
+runDropRemoteSchemaPermissions (DropRemoteSchemaPermissions name roleName) = do
+  remoteSchemaMap <- scRemoteSchemas <$> askSchemaCache
```

Comment:
I had thought quite a bit about this. I'm consciously not failing here when remote schema permissions are not enabled because I wanted to allow the user to be able to drop the permission, if somehow there are remote schema permissions (only the definitions in the DB)  even when remote schema permissions are not enabled. 

Please let me know if you feel differently about this.

Reply:
The above case can happen when the graphql-engine is run with remote schema permissions enabled and the user has added some remote schema permissions
and then the next time the graphql-engine is run it's run without enabling remote schema permissions.